.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
```

Comment:
Where's the docs? :grin: 

To document/steps:
- This is a convenience tool, not a safety measure: you need to explicitly install pre-commit if you want Git to warn you before you commit.
- Install the tool: `pip3 install pre-commit`
- Install the hook: `pre-commit install`

When committing something good, it shows the following output:
```
~/github/precice [add-pre-commit]$ git commit
clang-format.............................................................Passed
check xml............................................(no files to check)Skipped
check for merge conflicts................................................Passed
mixed line ending........................................................Passed
format preCICE configuration files...................(no files to check)Skipped
[add-pre-commit f886de24] test
 1 file changed, 1 insertion(+), 1 deletion(-)
```

Reply:
It's not only a convenience tool, you can use it to format your files using the correct version of clang-format, which is super useful if your distro doesn't provide it.

Reply:
Just FYI, this is thanks to a binary wheel distribution of clang-format via https://github.com/ssciwr/clang-format-wheel